x/evm/keeper/chainKeeper.go

```diff
@@ -383,7 +383,8 @@ func (k chainKeeper) getConfirmedDeposits(ctx sdk.Context) []types.ERC20Deposit
 func (k chainKeeper) GetConfirmedDepositsPaginated(ctx sdk.Context, pageRequest *query.PageRequest) ([]types.ERC20Deposit, *query.PageResponse, error) {
 	var deposits []types.ERC20Deposit
 
-	resp, err := query.Paginate(prefix.NewStore(k.getStore(ctx).KVStore, utils.KeyFromStr(confirmedDepositPrefix).AsKey()), pageRequest, func(key []byte, value []byte) error {
+	// TODO: refactor iteration over values using a prefix to avoid collisions
+	resp, err := query.Paginate(prefix.NewStore(k.getStore(ctx).KVStore, append(utils.KeyFromStr(confirmedDepositPrefix).AsKey(), []byte(key.DefaultDelimiter)...)), pageRequest, func(key []byte, value []byte) error {
```

Comment:
Adding a unit test for collision under this specific key seems a bit annoying to do but i can look more into it if you think it's worth it.